src/code/providers/post-message-provider.coffee

```diff
@@ -0,0 +1,31 @@
+ProviderInterface = (require './provider-interface').ProviderInterface
+getQueryParam = require '../utils/get-query-param'
+
+class PostMessageProvider extends ProviderInterface
+
+  @Name: 'postMessage'
+
+  constructor: (@options = {}, @client) ->
+    super
+      capabilities:
+        save: false
+        resave: false
+        export: if getQueryParam "saveSecondaryFileViaPostMessage" then 'auto' else false
+        load: false
+        list: false
+        remove: false
+        rename: false
+        close: false
+
+  canOpenSaved: -> false
+
+  save: (content, metadata, callback) ->
```

Comment:
It might be better to rename this `export` since that is the capability it is providing.

Reply:
I thought a little bit about that. I wanted to preserve the pre-existing notion that export/save are separate capabilities that can share an implementation, but as your comment suggests that implicit distinction was not entirely clear. With the latest commit I've made it explicit by adding the `saveAsExport()` method (I avoided `export` as it's a reserved word in JavaScript) along with a default implementation that defers to the `save()` method, so existing clients that support both capabilities but only implement `save()` will continue to work.